Java/core/src/main/java/com/amazon/randomcutforest/state/tree/AbstractNodeStoreMapper.java

```diff
@@ -50,12 +49,7 @@ public AbstractNodeStore toModel(NodeStoreState state, CompactRandomCutTreeConte
             cutValue = ArrayPacking.unpackFloats(state.getCutValueData(), capacity);
             leftIndex = ArrayPacking.unpackInts(state.getLeftIndex(), capacity, state.isCompressed());
             rightIndex = ArrayPacking.unpackInts(state.getRightIndex(), capacity, state.isCompressed());
-            if (state.isCanonicalAndNotALeaf()) {
-                reverseBits(state.getSize(), leftIndex, rightIndex, capacity);
-            } else {
-                replaceLeaves(leftIndex, capacity);
-                replaceLeaves(rightIndex, capacity);
-            }
+            reverseBits(state.getSize(), leftIndex, rightIndex, capacity);
```

Comment:
In the toState method of the same file, you have 

        boolean check = root != Null && root < capacity;
        state.setCanonicalAndNotALeaf(check);
        if (check) { // can have a canonical representation saving a lot of space

do you need the check? I didn't see you assigned any value to root so it will always be 0 and the check is always true.

Reply:
I agree -- I am worried that there may be version 2.0 models that needed these lines. But at the same time -- there has been so many changes it would be hard to guarantee any behavior. Let me remove these lines. For 2.0 models, as and when they arise, we should leverage the randomcutforest-serialize package and we do have a V2StateToV3ForestConverter -- but there are no tests (its unclear if any such model is in use even).

Reply:
The value assigned to the root would matter for the getParent() function. But we do not need the getParent() in the library currently (because we use the system stack cleverly :) ), but down the road it may be a good to have.

Reply:
The root is being saved by TreeMapper when it invokes the AbstractStoreMapper ... let me not change this and break things. But agreed that this may need clarification. The real core issue is that a tree and a node store and node view are all three different abstractions of an actual tree. Node view helps the visitor abstractions. Node store helps changing the byte, char, int in saving space and hence we are here ...